bundles/specmate-emfrest/src/com/specmate/emfrest/authentication/ProjectNames.java

```diff
@@ -0,0 +1,36 @@
+package com.specmate.emfrest.authentication;
+
+import javax.ws.rs.core.MultivaluedMap;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+import com.specmate.connectors.api.IProjectService;
+import com.specmate.emfrest.api.IRestService;
+import com.specmate.emfrest.api.RestServiceBase;
+
+@Component(service = IRestService.class)
+public class ProjectNames extends RestServiceBase {
+	public static final String SERVICE_NAME = "projectnames";
+	private IProjectService projectService;
+	
+	@Override
+	public String getServiceName() {
+		return SERVICE_NAME;
+	}
+	
+	@Override
+	public boolean canGet(Object obj) {
```

Comment:
you could restrict this to the root "/" by checking if obj is of type Resource

Reply:
Ok